src/error_handling.c

```diff
@@ -221,6 +224,13 @@ rcutils_set_error_state(
   gtls_rcutils_error_is_set = true;
 }
 
+rcutils_ret_t
+rcutils_set_error_code(rcutils_ret_t error_code){
+  gtls_rcutils_error_code = error_code;
+  gtls_rcutils_error_code_is_set  = true;
+  return error_code;
```

Comment:
I assume this is how it will be used. 
https://github.com/ros2/rmw_fastrtps/blob/1dcfe806874956158e1d93a77034c740e8790e19/rmw_fastrtps_shared_cpp/src/rmw_response.cpp#L150
```
return RMW_RET_TIMEOUT;
```
->
```
return RMW_SET_ERROR_CODE(RMW_RET_TIMEOUT);
```